core/picture.c

```diff
@@ -123,13 +123,13 @@ static bool dive_check_picture_time(const struct dive *d, timestamp_t timestamp)
  * The caller is responsible for actually adding the picture to the dive.
  * If no appropriate dive was found, no picture is created and NULL is returned.
  */
-struct picture *create_picture(const char *filename, int shift_time, bool match_all, struct dive **dive)
+struct picture *create_picture(const char *filename, offset_t shift_time, bool match_all, struct dive **dive)
```

Comment:
I really wonder if this is a valid assumption.

One of the "typical" cases with incorrect dates appears to be an incorrect "Unix epoch" date - so January 1970.
So yeah, that's 52 years.

Now, I don't know if the code would be broken in different ways for that case - but I just a few weeks ago stumbled across pictures that claimed to be from Dec 31 1969 (thanks, time zones)...